lib/Sema/SemaResolveIdentifier.cpp

```
@@ -265,6 +265,115 @@ void IdentifierResolver::visit(AsmFunctionDeclStmtType &FDS) {
+            Actions.addDecl(F);
+          break;
+        case Identifier::SpecialIdentifier::super_: {
+          auto Bases = DIR.getCurrentContract()->getResolvedBaseContracts();
+          if (Bases.size() > 1) {
+            Actions.Diag(Base->getLocation().getBegin(),
+                         diag::err_super_in_noninheritance_contract);
+          }
+          for (auto F : Bases[1]->getFuncs())
+            Actions.addDecl(F);
+        } break;
+        default:
+          assert(I && "Unimplement SpecialIdentifier Base");
+        }
+      } else if (auto C =
+                     dynamic_cast<ContractDecl *>(I->getCorrespondDecl())) {
+        // Inheritance call with Contract name
+        for (auto F : C->getFuncs())
+          Actions.addDecl(F);
+      } else if (auto CT = dynamic_cast<ContractType *>(I->getType().get())) {
+        // Contract external call
+        if (CT->getDecl()) {
+          for (auto F : CT->getDecl()->getFuncs())
+            Actions.addDecl(F);
+        } else if (I->getName() == "this") {
+          for (auto F : DIR.getCurrentContract()->getFuncs())
+            Actions.addDecl(F);
+        }
+      }
+    } break;
+    case Type::Category::Struct: {
+      if (auto ST = dynamic_cast<StructType *>(Base->getType().get())) {
+        auto Types = ST->getElementTypes();
+        auto ElementName = M.getName()->getName().str();
+
+        if (ST->hasElement(ElementName)) {
+          // Note: struct do not have sub-decl yet. check name only.
+          M.setType(Types[ST->getElementIndex(ElementName)]);
+          NameCheckByPass = true;
+          return;
+        } else {
+          Actions.Diag(M.getName()->getLocation().getBegin(),
+                       diag::err_no_member)
+              << ElementName << M.getBase()->getType()->getName();
+        }
+      }
+    } break;
+    case Type::Category::Address: {
+      // TODO: Handle Address case
+      // TODO: Address.send(x), should bypass send and check x.
```

Do we need to throw an unimplemented error here?